File: DALCore/Query/ClientAccountQuery.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Linq.Expressions;
using System.Text;
using System.Threading.Tasks;
using DALCore.Models;

namespace DALCore.Query
{
    public class ClientAccountQuery
    {
        public static Expression<Func<ClientAccount, bool>> GetNameQuery(string name)
        {
            if (string.IsNullOrWhiteSpace(name))
                return null/* TODO Change to default(_) if this is not a reference type */;
            return w => w.CompanyName.Contains(name) | w.Name.Contains(name);
        }
    }
}

File: DALNEW/Query/ClientAccountQuery.vb
Imports System.Linq.Expressions
Imports BO
Public Class ClientAccountQuery
    Public Shared Function GetNameQuery(name As String) As Expression(Of Func(Of ClientAccount, Boolean))
        If String.IsNullOrWhiteSpace(name) Then Return Nothing
        Return Function(w) w.CompanyName.Contains(name) Or w.Name.Contains(name)
    End Function
End Class
